src/api/utils/tradeQuoter.ts

```diff
@@ -95,12 +95,18 @@ export class TradeQuoter {
       fromAddress, [fromTokenAddress, toTokenAddress]
     );
 
+    // This value is already floored. Number that ends in 8, should end in .7
     const fromTokenRequestAmount = this.calculateFromTokenAmount(
       setOnChainDetails,
       fromTokenAddress,
       amount
     );
 
+    const positionForFromToken = setOnChainDetails
+      .positions
+      .find((p: any) => p.component.toLowerCase() === fromTokenAddress.toLowerCase());
```

Comment:
This is the true final amount that we want to submit for quote validation & return to the front-end, right?